proto/server.proto

```diff
@@ -0,0 +1,51 @@
+syntax = "proto3";
+
+package proto;
+
+// NotaryServer defines the TUF operations that the server supports
+service Notary {
+	rpc Auth(Token) returns (AuthResponse) {}
+	rpc Update(Updates) returns (UpdateResponse) {}
+	rpc Get(RepoRole) returns (Repository) {}
+	rpc GetKey(RepoRole) returns (KeyResponse) {}
+}
+
+// Token holds a JWT that the Notary server will check for permissions
+message Token {
+	bytes jwt = 1;
+}
+
+message Updates {
+	bytes jwt = 1;
+	string gun = 2;
+	Repository repo = 3;
+}
+
+message RepoRole {
+	bytes jwt = 1;
```

Comment:
Everything is open for suggestions. Hadn't looked deep enough into GRPC to find how it handles headers. Will look further.